src/caencoder.c

```diff
@@ -278,10 +281,49 @@ static int ca_encoder_node_read_device_size(CaEncoderNode *n) {
         if (n->fd < 0)
                 return -EBADFD;
 
-        if (ioctl(n->fd, BLKGETSIZE, &u) < 0)
+        r = read(n->fd, &magic, sizeof(magic));
```

Comment:
hmm, i think it would be much nicer to use a pread() here, and read the whole superblock at once (possibly in a union, together with other superblocks, if we later add support for them), so that we just have that one syscall instead of a read() + read() + seek() for the detection...